x-pack/elastic-agent/pkg/composable/controller.go

```diff
@@ -107,6 +114,10 @@ func (c *controller) Run(ctx context.Context, cb VarsCallback) error {
 			cancel()
 			return errors.New(err, fmt.Sprintf("failed to run provider '%s'", name), errors.TypeConfig, errors.M("provider", name))
 		}
+		// manually register providers that want to pass to the Vars as "fetch" providers
+		if _, ok := FetchContextProviders[name]; ok {
+			fetchContextProviders[name] = state.provider
```

Comment:
Another option here instead of adding a new Register function just for Fetching providers, you could check if a `ContexProvider` implements the `FetchContextProvider` interface and then set them into the `fetchContextProviders mapping.

Reply:
I will pick this option.